src/auth/reducer.ts

```diff
@@ -1,25 +1,75 @@
 import { createSlice } from '@reduxjs/toolkit';
+import type { User } from 'oidc-client-ts';
 
 import {
   FETCH_API_TOKEN,
   TOKEN_NOT_FOUND,
   RECEIVE_API_TOKEN,
   ReceiveApiTokenAction,
+  USER_EXPIRED,
+  USER_FOUND,
+  UserFoundAction,
+  SILENT_RENEW_ERROR,
+  SESSION_TERMINATED,
+  LOADING_USER,
+  USER_SIGNED_OUT,
 } from './types';
 
+type OidcState = {
+  user: User | null;
+  isLoadingUser: boolean;
+};
+
 type CurrentDisplayState = {
   apiToken: string;
   isFetching: boolean;
 };
 
-const initialState: CurrentDisplayState = {
+const initialOidcState: OidcState = {
+  user: null,
+  isLoadingUser: false,
+};
+
+const oidcSlice = createSlice({
+  name: 'oidc',
+  initialState: initialOidcState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(USER_EXPIRED, (state) => {
+        state.user = null;
```

Comment:
I don't like duplicate code, but I think in this case we cannot avoid it. At least I don't know how to get rid of duplicating `state.user = null` and `state.isLoadingUser = false` statements. Making a function for setting them doesn't actually make the code shorted nor easier to read 😅

Reply:
Could make a function that receives the state, which would remove one line, but not sure if it really improves things